common/app/common/Logging.scala

```diff
@@ -25,6 +29,27 @@ trait Logging {
   def logErrorWithCustomFields(message: String, error: Throwable, customFields: List[LogField]): Unit = {
     log.logger.error(customFieldMarkers(customFields), message, error)
   }
+
+  // Transparent error logging on exceptions: log context and exception on error, and pass on the exception
+  def errorLoggingF[A](context: String)(task: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    def errorLoggingFuture(f: Future[A]): Future[A] = {
+      f.recover {
+        case NonFatal(e) => Logger.error(context, e) ; throw e
+      }
+    }
+
+    Try(task) match {
```

Comment:
I am not sure I understand why the `Future` is being wrapped in a `Try`

Reply:
It's subtle :)

It's possible for `task: => Future[A]` to throw an exception which is not part of a Future.

Take this contrived example...
```
task = { throw new Exception() ; Future("Hi") }
```
future.recover wouldn't log the exception thrown here since the exception is thrown outside the future